app/Models/WaitingRoom.php

```diff
@@ -88,11 +88,15 @@ public function getDefaultSplit()
 
     /*
      * Creates competitions based on the given user split.
+     *
+     * @param \Gladiator\Models\Contest $contest
+     * @param Array                     $split
+     * @param \Illuminate\Http\Request  $request
      */
-    public function saveSplit($contest, $split, $endDate)
+    public function saveSplit($contest, $split, $request)
     {
         $startDate = Carbon::now()->startOfDay();
-        $endDate = new Carbon($endDate);
+        $endDate = new Carbon($request->competition_end_date);
```

Comment:
It might make sense to use the Request's array-access interface here `$request['competition_end_date']` so that you could use an array for this parameter if you ever wanted to initiate this from somewhere other than a web request.